modules/carto/src/api/maps-v3-client.js

```diff
@@ -167,6 +183,70 @@ export async function getData({type, source, connection, credentials, geoColumn,
   }
 
   const {accessToken} = localCreds;
+  return {url, mapFormat, accessToken};
+}
 
-  return await request({url, format: mapFormat, accessToken});
+/* global setInterval, URLSearchParams */
+async function _getMapDataset(dataset, accessToken) {
+  // First fetch metadata
+  const {connectionName: connection, source, type} = dataset;
+  const {url, mapFormat} = await _getDataUrl({
+    credentials: {accessToken},
+    connection,
+    source,
+    type
+  });
+
+  // Extract the last time the data changed
+  const cache = parseInt(new URLSearchParams(url).get('cache'), 10);
+  if (cache && dataset.cache === cache) {
+    return false;
+  }
+  dataset.cache = cache;
+
+  // Only fetch if the data has changed
+  const data = await request({url, format: mapFormat, accessToken});
+  dataset.data = data;
+  return true;
+}
+
+async function getMapDatasets({datasets, publicToken}) {
+  const promises = datasets.map(dataset => _getMapDataset(dataset, publicToken));
+  return await Promise.all(promises);
+}
+
+export async function getMap({mapId, credentials, autoRefresh, onNewData}) {
+  const localCreds = {...getDefaultCredentials(), ...credentials};
+
+  log.assert(mapId, 'Must define map id');
+
+  log.assert(localCreds.apiVersion === API_VERSIONS.V3, 'Method only available for v3');
+  log.assert(localCreds.apiBaseUrl, 'Must define apiBaseUrl');
+  if (!localCreds.mapsUrl) {
+    localCreds.mapsUrl = buildMapsUrlFromBase(localCreds.apiBaseUrl);
+  }
+
+  if (autoRefresh || onNewData) {
+    log.assert(onNewData, 'Must define `onNewData` when using autoRefresh');
+    log.assert(typeof onNewData === 'function', '`onNewData` must be a function');
+    log.assert(autoRefresh > 0, '`autoRefresh` must be a positive number');
+  }
+
+  const url = `${localCreds.mapsUrl}/public/${mapId}`;
+  const map = await request({url});
```

Comment:
Would be great if we could cache here the status response to say this functionality only works for public maps